core/src/main/java/hudson/util/ProcessTree.java

```diff
@@ -1727,7 +1727,7 @@ public synchronized EnvVars getEnvironmentVariables() {
 
             @Override
             @NonNull
-            public List<String> getArguments() {
+            public synchronized List<String> getArguments() {
```

Comment:
`getEnvironmentVariables()` is synchronized, so this should be as well. Not to mention every other platform's `getArguments()` in this class is also synchronized. This seems like an oversight.